src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -2867,22 +2850,41 @@ public function doSelect(?ConnectionInterface $con = null)
 
         $params = [];
         $sql = $this->createSelectSql($params);
+        $stmt = null;
         try {
             $stmt = $con->prepare($sql);
             $db->bindValues($stmt, $params, $dbMap);
             $stmt->execute();
         } catch (Exception $e) {
-            if (isset($stmt)) {
-                $stmt = null; // close
-            }
-            Propel::log($e->getMessage(), Propel::LOG_ERR);
-
-            throw new PropelException(sprintf('Unable to execute SELECT statement [%s]', $sql), null, $e);
+            $this->handleStatementException($e, $stmt, $sql);
         }
 
         return $con->getDataFetcher($stmt);
     }
 
+    /**
+     * Logs an exception and adds the complete SQL statement to the exception.
+     *
+     * @param \Exception $e
+     * @param \Propel\Runtime\Connection\StatementInterface|\PDOStatement $stmt
+     * @param string|null $sql
+     *
+     * @throws \Propel\Runtime\Exception\PropelException
+     *
+     * @return void
+     */
+    protected function handleStatementException(Exception $e, $stmt, ?string $sql = '')
+    {
+        if (is_a($stmt, StatementWrapper::class)) {
+            $sql = $stmt->getExecutedQueryString();
+        }
+        $initialMessage = $e->getMessage();
+        Propel::log($initialMessage, Propel::LOG_ERR);
+        $newMessage = sprintf("Unable to execute statement [%s]\nReason: [%s]", $sql ?? '', $initialMessage);
```

Comment:
We can use single quotes here.

Reply:
Then the '\n' wouldn't work. But I changed it so that it doesn't use sprintf anymore

Reply:
No, the sprintf was fine, I didnt see the \n usage. All good either way.